dds/DCPS/DataSampleHeader.h

```diff
@@ -260,6 +260,11 @@ struct OpenDDS_Dcps_Export DataSampleHeader : public PoolAllocationBase {
     }
   }
 
+  ACE_Time_Value get_source_timestamp() const
```

Comment:
Return a `SystemTimePoint`?

Reply:
I thought it might be more flexible to have it return ACE_Time_Value in case the user needs a MonotonicTimePoint rather than SystemTimePoint? But maybe that's an unlikely use case...

Reply:
No this should always be system time:

> `SystemTimePoint` should be used when dealing with the DDS API and timestamps on incoming and outgoing messages.

Also if there's a situation in OpenDDS code where a `ACE_Time_Value` can be either be system or monotonic then something is wrong with the design because it could break the kind of guarantees that the time types are trying to ensure. That's why I also say that we should only deal with `ACE_Time_Value` directly when interacting with ACE.

Reply:
Gotcha, ok, I'll change it to SystemTimePoint.